### PR TITLE
Set type:module

### PR DESCRIPTION
**Pull request by tinydylan**
Other changes:
- Running mocha by specifying ts files, not js.
- Junit and console output for mocha
- Package upgrades

### CODE REVIEW

**Comment by tinydylan** [comment]
@tiny-james I think this does the trick? Not sure. Tests seem to pass.

**Review by tiny-james — APPROVED** [review]
Also Lee suggested in the slack channel that we update walugi in the pipeline to a newer version, possibly 3.3.0.

**Comment by tinydylan** [comment]
> Also Lee suggested in the slack channel that we update walugi in the pipeline to a newer version, possibly 3.3.0.

Thanks. Done.

File: package.json [hunk 1]
```diff
@@ -6,15 +6,17 @@
   "module": "./lib/main/ts/api/Main.js",
   "scripts": {
     "prepublishOnly": "tsc",
-    "build": "tsc && tslint --project .",
-    "test": "yarn build && mocha -r esm lib/test/**/*Test.js"
+    "lint": "tslint --project .",
+    "build": "tsc",
+    "test": "yarn build && yarn lint && mocha -r esm --require ts-node/register --require source-map-support/register --recursive --reporter mocha-multi-reporters --reporter-options configFile=mocha-multi-reporters-config.json src/test/ts/**/*Test.ts"
```
Comment (lnewson) [inline package.json]: That's one hell of a command :joy:

File: yarn.lock [hunk 2]
```diff
@@ -415,6 +437,11 @@ locate-path@^3.0.0:
     p-locate "^3.0.0"
     path-exists "^3.0.0"
 
+lodash@^4.16.4:
```
Comment (lnewson) [inline yarn.lock]: Looks like you need to run `npx yarn-deduplicate` here? As there are 2 lodash versions now.

**Review by lnewson — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ashfordneil) [reply package.json]: Can we pull the CLI arguments out into a "mocha" field in the package.json?

**Review by ashfordneil — APPROVED** [review]

File: yarn.lock (hunk 2 above)
Comment (tinydylan) [reply yarn.lock]: Fixed.

File: package.json (hunk 1 above)
Comment (tinydylan) [reply package.json]: I've pulled it out into a `.mocharc.yml` file. Is that ok?

Comment (lnewson) [reply package.json]: Much nicer thanks!

**Pull request merged**